metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/config/SpringWebConfig.java

```diff
@@ -150,6 +151,11 @@ private <K, V> Map<K, V> concat(Supplier<Map<K, V>> a, Supplier<Map<K, V>> b) {
         : b.get() == null
             ? a.get()
             : Stream.concat(a.get().entrySet().stream(), b.get().entrySet().stream())
-                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+                .collect(
+                    Collectors.toMap(
+                        Map.Entry::getKey,
+                        Map.Entry::getValue,
+                        (v1, v2) -> v1,
```

Comment:
in this case, `v1` is `oComponents` and `v2` is `v3Components`

Shouldn't we always prefer `v2` then?

Reply:
actually im not sure if that order is guaranteed between the two variables in the lambda in this manner 
I think we should process `v3Components` first always because those are explicitly built by the `OpenAPIV3Generator`
`Stream.concat(b.get().entrySet().stream(), a.get().entrySet().stream())`

Reply:
This assumes that the component is same from both. So v1 or v2 does not matter. Instead of throwing an error, the logic chooses one.

Reply:
Right but in the future if there are conflicts by name, I think it is safer to default to the v3 generator component since that is where we have the logic to construct the spec

Reply:
ok. WIll make the change..